src/BaseModel.ts

```diff
@@ -111,7 +108,7 @@ export class BaseModel {
    * @param fields Container fields
    * @param source Container data source
    */
-  public addContainer (name: string, fields: any, source: any = null): BaseModel {
+  public addContainer ({ name, fields, source }: ContainerBase): BaseModel<Parent> {
```

Comment:
I was hoping to avoid this style of creating a new container (object-like. it's gonna be dirty when you will be adding a container in you app)

Reply:
Will you accept it?

Reply:
Maybe you've got another solution in mind? The other stuff is very decent tho